lib/storage/src/content_manager/consensus_state.rs

```diff
@@ -91,9 +94,29 @@ impl<C: CollectionContainer> ConsensusState<C> {
             consensus_thread_status: RwLock::new(ConsensusThreadStatus::Working {
                 last_update: Utc::now(),
             }),
+            message_send_failures: Default::default(),
         }
     }
 
+    pub fn record_message_send_failure<E: Error>(&self, peer_address: &Uri, error: E) {
+        let mut message_send_failures = self.message_send_failures.write();
+        let mut entry = message_send_failures
+            .entry(peer_address.to_string())
+            .or_insert_with(Default::default);
+        // Log only first error
+        if entry.count == 0 {
+            log::error!("Failed to send message to {peer_address} with error: {error}")
+        }
+        entry.count += 1;
+        entry.latest_error = Some(error.to_string());
```

Comment:
If I understand properly, this always increments the count and overwrite the `latest_error`.
So the error could be different at each increment right?